core/app/models/spree/gallery/product_gallery.rb

```diff
@@ -0,0 +1,16 @@
+module Spree
+  module Gallery
+    class ProductGallery
+      def initialize(product)
+        @product = product
+      end
+
+      # A list of all images associated with this gallery
+      #
+      # @return [Enumerable<Spree::Image>] all images in the gallery
+      def images
```

Comment:
I think we would gain much more isolation from  this class if it would have an `image_urls` method that takes a `size:` parameter.